server/PythonClient/Client.py

```diff
@@ -0,0 +1,339 @@
+
+    @staticmethod
+    def readDouble(reader: RecieveFunction):
+        struct.unpack('f', reader(8))
+
+    @staticmethod
+    def readBytesToString(reader: RecieveFunction, nBytes):
+        return reader(nBytes).decode(ENCODING)
+
+# RESPONSIBLE FOR READING DATA FROM SERVER
+
+
+class FractalReader:
+
+    def __init__(self):
+        super().__init__()
+
+    def read(self, reciever: RecieveFunction):
+        # Length of paylaod in bytes
+        payloadlen = StreamHelper.readInt(reciever)
+
+        # Read payload name length
+        payloadNameLen = StreamHelper.readInt(reciever)
+        # Read payload name
+        payloadName = StreamHelper.readBytesToString(
+            reciever, payloadNameLen)
+
+        # Read segments length
+        segmentsLen = StreamHelper.readInt(reciever)
+        # Read segments
+        segments = StreamHelper.readBytesToString(
+            reciever, segmentsLen)
+
+        # Read JSON payload length
+        jsonLen = StreamHelper.readInt(reciever)
+        # Read JSON payload
+        jsondata = StreamHelper.readBytesToString(
+            reciever, jsonLen)
+
+        # parse JSON payload data
+        payloadData = json.loads(jsondata)
+
+        # parse segment JSON payload data
+        parsed_segments = json.loads(segments)
+
+        return RecievedPayload(payloadData, self.readSegments(reciever, parsed_segments))
+
+    def readSegments(self, reciever, parsed_segments):
+        segmentDict = {}
+        for segment in parsed_segments:
+            segmentKey = next(iter(segment.keys()))
+            segmentMeta = next(iter(segment.values()))
+            size = int(segmentMeta["size"])
+            file_name = ""
+            if Segment._META_KEY_FILE_NAME in segmentMeta:
+                file_name = segmentMeta[Segment._META_KEY_FILE_NAME]
+            else:
+                file_name = str(uuid.uuid1())
+
+            fileFullPath = TMP_DIR + file_name
+            f = open(TMP_DIR + file_name, 'wb')
+            buffer = 25
+            remaining = size
+            l = reciever(min(buffer, remaining))
+            while (l):
+                remaining -= len(l)
+                f.write(l)
+                l = reciever(min(buffer, remaining))
+            f.close()
+            segmentDict[segmentKey] = RecivedSegment(segmentMeta, fileFullPath)
+        return segmentDict
+
+
+class RecivedSegment:
+    def __init__(self, segmentMeta, fileName):
+        self.segmentMeta = segmentMeta
+        self.fileName = fileName
+
+
+class RecievedPayload:
+    def __init__(self, payloadData, segments):
+        self.payloadData = payloadData
+        self.segments = segments
+
+
+class Segment(metaclass=abc.ABCMeta):
+
+    _META_KEY_SIZE = "size"
+    _META_KEY_MIME_TYPE = "mime_type"
+    _META_KEY_FILE_NAME = "file_name"
+
+    def __init__(self):
+        super().__init__()
+        self.segmentMeta = {}
+
+    @ abc.abstractmethod
+    def writeToStream(self):
```

Comment:
burde vel ha samme signatur som implementasjonene
```suggestion
    def writeToStream(self, sendFunction: SendFunction):
```